### DISCUSSION TITLE
How to transfer camera focal lenght to fov params in pytorch3d?

### DISCUSSION THREAD

**Discussion by lucasjinreal** [facebookresearch/pytorch3d#1035] [Q&A]
In pytorch3d , it only have fov params:

```
T = T.unsqueeze(0).repeat(BS, 1)
                new_cam = FoVPerspectiveCameras(zfar=1000, znear=0.05,
                    R=self.default_R, T=T, fov=self.fov, device=self.device)
```

However, I just got focal lenght params, how to send it to camera as an equal result with fov?

**Comment by bottler** [comment]
In general, for square images, you have  `1/(focal length in ndc)` equal to `tan(0.5*fov)`.

**Reply by lucasjinreal** [reply]
@bottler what about none-square images? In more general way to ask this question, if I have a human mesh model which trained on the hyposis of focal_len = 50, then how should I visiualize this model prediction output with fov params?

my input image could be rectangle.

**Reply by lucasjinreal** [reply]
Or, does there any other Camera class except this FoVPerspectiveCameras which can take focal_len as camera params?

**Reply by bottler** [reply]
Yes. If you don't want to provide fov, you can use PerspectiveCameras.
